en/components/toggle.md

```diff
@@ -95,35 +112,83 @@ Then in the template we need to declare **IgxToggleActionDirective** by assignin
 
 ```html
 <!--template.component.html-->
-<button [igxToggleAction]="toggle">Open/Close</button>
-<div igxToggle #toggle="toggle">
-    <p>Content</p>
+<button class="toggle-button"  igxButton="raised" [igxToggleAction]="toggleRef">Toggle</button>
+<div class="toggle-content" igxToggle #toggleRef="toggle">
+    <section class="toggle-section">
+        <img src="link to random image"/>
+    </section>
 </div>
 ```
 
 After these changes the toggle should work exactly in the same way.
 
-<div class="sample-container" style="height: 400px">
-Demo
+<div class="sample-container loading" style="height: 370px">
+    <iframe id="toggle-sample-2-iframe" src='{environment:demosBaseUrl}/toggle-sample-2' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
 </div>
+<div>
+    <button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="toggle-sample-2-iframe" data-demos-base-url="{environment:demosBaseUrl}">                       view on stackblitz
+    </button>
+</div>
+
+### Automatic toggle actions with service provider
+
+There is a convenient way to keep the state and command our **igxToggle** through **igxNavigationService** provider which would be convenient for us to keep tracking it everywhere in our applicaiton. We just need to declare identifier to our **igxToggle** directive and the registration happens for us. If we would like this state to be controlled automatically again we just need to pass this identifier to our helping **igxToggleActionDirective**.
+
+```typescript
+// app.module.ts
+
+...
+import { IgxToggleModule, IgxButtonModule } from 'igniteui-angular/main'
+
+@NgModule({
+    ...
+    imports: [..., IgxToggleModule, IgxButtonModule]
+    ...
+})
+export class AppModule {}
+```
+
+```html
+<!--template.component.html-->
+<button igxToggleAction="toggleId" [closeOnOutsideClick]="true" class="toggle-button"  igxButton="raised">Toggle</button>
+<div igxToggle id="toggleId" class="toggle-content" [collapsed]="false">
+    <section class="toggle-section">
+    <h3>Click 
+        <br/> Out of the Box</h3>
+    </section>
+</div>
+```
+
+If all went well we have to see this:
```

Comment:
"we have to see this" means that something is so amazing that we have no choice but to view it.  I think it's better as "If all went well, it will look like this:".